src/en/robust-toolbox/ecs.md

```diff
@@ -110,26 +110,25 @@ As you can see, by using composition all our problems from before are solved nea
-`StackSystem` has a few methods to operate on `StackComponent`, and change its values. So to use a certain amount of things on the stack, you use `StackSystem.Use`, to split it you use `StackSystem.Split`, etc, and `StackSystem` will take care of everything for you. Because turns out that changing the stack amount value isn't enough. You also need to do funny stuff such as: dirtying the component for network syncing purposes, setting the appearance value, raise a `StackCountChanged` event, etc.
-The way you would do this in E/C would be to put all of that logic in the "amount" property on `StackComponent`, or maybe a method. Then you could `private` the actual "amount" number away. However the E/C architecture will not be accepted in this codebase, we will only use the ECS architecture going forward.
+But perhaps that's the solution? Mobs can have machine bits, and machines can have mob bits, so just create an abstract machine mob class! But at the same time, that wouldn't make much sense either! Most mobs will not need "power consumption features", and few machines will be player-controlled at all.
 
-See, putting any kind of logic in a component class doesn't make sense at all. If we think about how stuff in ECS is structured, it's like this:
-1. Our game world has a bunch of entities, the entities have components
-2. There are systems that operate on components
+To solve this difficult design decision, we would humbly propose an alternative design perspective. Instead of inheriting features, what if we composed them? Think of it as building legos instead of branching trees, but how would this work? Suppose you want an entity to have hands, add a `HandsComponent`. If you want to make it consume power, `PowerConsumerComponent` will help! If you want to make it player-controllable, add `MindComponent`. What have we created? An entity with hands, power, and can be controlled by a player. We've made a "cyborg mob" out of reusable, generic components! 
 
-Therefore, components should only have data in them, and no logic whatsoever. Systems should give the entities their behavior, when they have the appropriate components.
-I'm more saying like... Rather than have logic in components change stuff, we want entity systems to operate and modify components. So instead of having:
-`HandsComponent.Pickup(ItemComponent)`, you would have:
-`HandsSystem.Pickup(UserEntity, ItemEntity)`.
+Inheritance may be perfectly fine in nuanced circumstances, such as small, self-contained inheritance trees (see something like `SoundSpecifier` for example). However, at scale its benefits become detriments quickly. Especially when you have a game as large and complex as ss14.
```

Comment:
It's not really about scale, it's about the object-oriented model running into problems modeling in-game data objects. It may be helpful to link external resources, too.